src/de.rs

```diff
@@ -117,12 +114,13 @@ impl<'de, R: dec::Read<'de>> Deserializer<R> {
     where
         V: Visitor<'de>,
     {
-        let tag_value = pull_one(&mut self.reader)?;
-        match tag_value {
+        let tag = dec::TagStart::decode(&mut self.reader)?;
+
+        match tag.0 {
             CBOR_TAGS_CID => visitor.visit_newtype_struct(&mut CidDeserializer(self)),
             _ => Err(DecodeError::TypeMismatch {
                 name: "CBOR tag",
-                byte: tag_value,
+                byte: tag.0 as u8,
```

Comment:
Adding a new enum variant for "UnsupportedTag" may be better. Otherwise, we're truncating the tag here and that could be very confusing (e.g., if the bottom byte is 42).

Reply:
Works for me.